### PR TITLE
[WIP]Added bulk of the test driven development lab. Still in Draft stages.

### PR DESCRIPTION
**Pull request by Luke1298**
This request has code that is a work in progress. It should not be merged into develop.

### CODE REVIEW

**Comment by Luke1298** [comment]
@shanemcq18 Could you take a look at the solutions_ANSWER file and comment on the length of that one problem. (That is only a viable solution to problem 4 but it is quite lengthy) Do you think it is manageable?

File: Labs/UnitTestingInPython/UnitTestingInPython.tex [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+\end{tikzpicture}\\
+The idea must be a bit more concrete then "I want to add a chat room to my website." The general idea of how the chat room will look in code (like a history module, a data base, and a sketch of required functions, etc.) is required.
+Next that idea is transformed into tests because the inputs and outputs of required functionality are understood.
+Following the example of the chat room you could write a unit test that checks what a given api returns when a user has broadcasted some message.
+The implementation is simply adding, changing and editing code until all the required tests pass.
+
+\begin{problem}
+Use TDD to create a module which takes a infix polynomail (of one arbitary variable) and plots it. For example it should take in:
+"$5x^4 + 4x^3 + 3x^2+ 2x + 1$" and plot:
+Equivlently "$5y^4 + 4y^3 + 3y^2+ 2y + 1$" (you may throw an excpetion if two variables are given).
```
Comment (tannert) [inline Labs/UnitTestingInPython/UnitTestingInPython.tex]: 'Equivalently' is spelled wrong

File: Labs/UnitTestingInPython/UnitTestingInPython.tex [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+\draw[->, thick] (id) -- (ts);
+\draw[->, thick] (ts) -- (im);
+\end{tikzpicture}\\
+The idea must be a bit more concrete then "I want to add a chat room to my website." The general idea of how the chat room will look in code (like a history module, a data base, and a sketch of required functions, etc.) is required.
+Next that idea is transformed into tests because the inputs and outputs of required functionality are understood.
+Following the example of the chat room you could write a unit test that checks what a given api returns when a user has broadcasted some message.
+The implementation is simply adding, changing and editing code until all the required tests pass.
+
+\begin{problem}
+Use TDD to create a module which takes a infix polynomail (of one arbitary variable) and plots it. For example it should take in:
```
Comment (tannert) [inline Labs/UnitTestingInPython/UnitTestingInPython.tex]: 'Polynomial' is spelled wrong

Comment (Luke1298) [inline Labs/UnitTestingInPython/UnitTestingInPython.tex]: Dude!!! Using WIP I love it!!! Working for the dev team was totally worth it if it got you using that :)

**Pull request closed without merging**